nginx.conf

```diff
@@ -0,0 +1,130 @@
+## NEED CHANGE (defines the user of the nginx workers)
+# user user group;
+
+## THIS PARAMETERS BE SAFELY OVER RIDDEN BY YOUR DEFAULT NGINX CONF
+worker_processes 1;
+env THREESCALE_DEPLOYMENT_ENV;
+env THREESCALE_PORTAL_ENDPOINT;
+env THREESCALE_CONFIG_FILE;
+
+# error_log stderr notice;
+# error_log logs/error.log warn;
+
+events {
+    worker_connections 256;
+}
+
+http {
+  lua_shared_dict api_keys 10m;
+  lua_shared_dict configuration 10m;
+  lua_shared_dict locks 1m;
+  server_names_hash_bucket_size 128;
+
+  error_log stderr info;
+  access_log /dev/stdout combined;
+
+  lua_package_path ";;${prefix}?.lua;${prefix}src/?.lua";
+
+  # Enabling the Lua code cache is strongly encouraged for production use. Here it is enabled by default for testing and development purposes
+  lua_code_cache off;
+
+  init_by_lua_block {
+    require('luarocks.loader')
+  }
+
+  resolver 127.0.0.1;
+
+  server {
+    listen 8090;
+
+    server_name _;
+
+    include conf.d/management.conf;
+  }
+
+  server {
+    listen 8081;
+
+    server_name backend;
+
+    include conf.d/backend.conf;
+  }
+
+  server {
+    listen 8080;
+
+    server_name _;
+    underscores_in_headers on;
+
+    set_by_lua $deployment 'return os.getenv("THREESCALE_DEPLOYMENT_ENV");';
+
+    location = /threescale_authrep {
+      internal;
+
+      set $backend_host 'backend';
+      set $backend_endpoint 'http://127.0.0.1:3001';
+
+      proxy_pass_request_headers off;
+      proxy_pass $backend_endpoint/transactions/authrep.xml?$backend_authentication_type=$backend_authentication_value&service_id=$service_id&$usage&$credentials; # &log%5Bcode%5D=$arg_code&log%5Brequest%5D=$arg_req&log%5Bresponse%5D=$arg_resp;
+      proxy_set_header  Host  "$backend_host";
+      proxy_set_header  X-3scale-User-Agent "nginx$deployment";
+      proxy_set_header  X-3scale-Version "2016-08-29T08:09:01Z";
+
+      log_by_lua_block {
+        ngx.log(ngx.INFO, '[authrep] ' .. ngx.var.request_uri .. ' ' .. ngx.var.status)
+      }
+  }
+
+
+    location = /out_of_band_authrep_action {
+      internal;
+
+      proxy_pass_request_headers off;
+
+      content_by_lua_block { require('provider').post_action_content() }
+    }
+
+
+    location / {
+      set $provider_key null;
+      set $cached_key null;
+      set $credentials null;
+      set $usage null;
+      set $service_id null;
+      set $proxy_pass null;
+      set $secret_token null;
+      set $resp_body null;
+      set $resp_headers null;
+      set $backend_authentication_type null;
+      set $backend_authentication_value null;
+
+      proxy_ignore_client_abort on;
+
+      access_by_lua_block {
+        local configuration = require('configuration')
+        local provider = require('provider')
+
+        local config = configuration.boot()
+
+        provider.init(config)
+        provider.access()
+      }
+
+      body_filter_by_lua_block {
```

Comment:
Is there some way to make these optional so we're not always logging requests? For some clients this can increase their latency quite a bit and is a common bit of code that we have had to rip out in the past